IDE/ECLIPSE/DEOS/deos_malloc.c

```diff
@@ -19,27 +19,11 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1335, USA
  */
 #include <wolfssl/wolfcrypt/settings.h>
-#include <wolfssl/ssl.h>
-
-#define ROUND_UP(x, align)  (((int) (x) + (align - 1)) & ~(align - 1))
-#define SIZEOF_HEADER sizeof(size_t)  /* tracks size of allocated block */
-
-#define HEAP_SIZE_MAX   (1*1024*1024)
-
-static size_t allocatedMemory = 0;
-
-size_t getMemAllocatedSize_deos(size_t* size){
-
-    if (size)
-        *size = allocatedMemory;
-
-    return allocatedMemory;
-}
-
-/* Simply returns without freeing any memory. */
+#include <wolfssl/wolfcrypt/types.h>
+#include <deos.h>
 
 void free_deos(void *ptr) {
-    //printf("fake free_deos()\n");
```

Comment:
malloc and free was unavailable with Deos DO-178.  Memory is expected to be managed by an application. That's why we have this file. You should consider wrapping the code with WOLFSSL_NO_MALLOC or another flag instead of removing it.

Reply:
You can remove the code block. Deos confirmed malloc/free have been added in the ansi library at version 4.7 sometime last year.  The ansi library is verified to DO-178 DAL A.